### PR TITLE
feat(discover) Use new discover API in discover2 endpoints

### PR DESCRIPTION
**Pull request by markstory**
Use the new discover API methods in each of the discover related endpoints. I've done them all at once to minimize potential inconsistencies that users will see. This does increase the potential risk as it is a flash cut. I'm planning on using staging to verify these changes with any saved queries our early access group has.

### CODE REVIEW

File: src/sentry/api/bases/organization_events.py [hunk 1]
```diff
@@ -11,29 +11,40 @@
...
+            return orderby
+
+    def reference_event(self, request, organization):
+        fields = request.GET.getlist("field")[:]
+        reference_event_id = request.GET.get("referenceEvent")
+        if reference_event_id:
+            return ReferenceEvent(organization, reference_event_id, fields)
+
+    def get_snuba_query_args(self, request, organization, params):
+        filter = self.get_snuba_filter(request, organization, params)
```
Comment (wmak) [inline src/sentry/api/bases/organization_events.py]: Should we avoid using `filter` since its a builtin?

**Comment by dashed** [comment]
@markstory  I tried out this PR and generated errors for these URLs just from loading the discover landing pag:

http://dev.getsentry.net:8000/api/0/organizations/sentry/events-stats/?interval=30m&query=event.type%3Atransaction&statsPeriod=14d

http://dev.getsentry.net:8000/api/0/organizations/sentry/events-stats/?end=2019-10-23T23%3A50%3A09&interval=30m&query=event.type%3Atransaction&start=2019-10-21T07%3A49%3A22

Screen Shot 2019-12-13 at 12 15 11 AM
Screen Shot 2019-12-13 at 12 15 17 AM

**Comment by markstory** [comment]
That sounds like your snuba is out of date. Try restarting your devservices.

**Comment by dashed** [comment]
Seems like the issue went away after restarting devservices.

File: src/sentry/static/sentry/app/components/tagDistributionMeter/index.jsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ export default class TagDistributionMeter extends React.Component {
 
           const tooltipHtml = (
             <React.Fragment>
-              <div className="truncate">{value.name}</div>
+              <div className="truncate">{value.name || t('n/a')}</div>
```
Comment (dashed) [inline src/sentry/static/sentry/app/components/tagDistributionMeter/index.jsx]: @markstory The proptypes for `segments.name` and `segments.value` will need to be updated to have `isRequired` removed.

Comment (markstory) [reply src/sentry/static/sentry/app/components/tagDistributionMeter/index.jsx]: Done.

**Review by leedongwei — APPROVED** [review]

**Pull request merged**